iree/compiler/Dialect/Flow/Transforms/GenerateDispatchTags.cpp

```diff
@@ -28,28 +27,38 @@ namespace Flow {
 namespace {
 
 //Generate IR at executable scope.
-static std::string executableIRString(Operation *op) {
-  auto function = dyn_cast<FuncOp>(op);
-  std::string execIR;
-  llvm::raw_string_ostream out(execIR)
-  // Print the function name and a newline before the Module.
-  out << " (function: " << function.getName() << ")\n";
-  function.getParentOfType<ModuleOp>().print(out);
-
-  // Print a newline before the IR.
-  out << "\n";
+static std::string getIRString(Operation *op) {
+  std::string IR;
+  llvm::raw_string_ostream out(IR);
+  // Print at module scope.
+  out << "  ('" << op->getName() << "' operation";
+  if (auto symbolName =
+          op->getAttrOfType<StringAttr>(SymbolTable::getSymbolAttrName())) {
+    out << ": @" << symbolName.getValue();
+  }
+  out << ") //----- //\n";
 
-  // Print the given function.
-  if (function) {
-    function.print(out);
-    return;
+  // Find the top-level operation.
+  auto *topLevelOp = op;
+  while (auto *parentOp = topLevelOp->getParentOp()) {
+    topLevelOp = parentOp;
+    out << "  ('" << topLevelOp->getName() << "' operation";
+    if (auto symbolName =
+            topLevelOp->getAttrOfType<StringAttr>(SymbolTable::getSymbolAttrName())) {
+      out << ": @" << symbolName.getValue();
+    }
+    out << ") //----- //\n";
   }
 
-  // Print the given module.
-  assert(isa<ModuleOp>(op) && "unexpected IR unit");
-  cast<ModuleOp>(op).print(out);
   return out.str();
 }
+//Create mlir::Attribute for source code.
+static mlir::Attribute getStringAsAttr(std::string str) {
+  const void * pointer = str.c_str();
+  return Attribute::getFromOpaquePointer(pointer);
+}
```

Comment:
I think you can use `Builder::getStringAttr()` instead of this (the `void*` use is super sketchy here :P)

Reply:
Haha! Super sketchy, I'll revise that. Once I have what's generated here passed down to Tracy correctly I can go back and work on replacing the IR dump with a useful summary.
Thanks for all the advice. (Deleted old comment because I quickly resolved that issue and ran into something more manageable).

Reply:
Hey, I'm having issues running iree-translate; keep getting segfaults when trying to reference the string attribute during the SerializeExecutables pass.
I can't tell whether it's minor tweaks needed or if I'm missing some major pieces.
Right now I have the GenerateDispatchTags pass setting a "sourceIRSummary" string attribute to IREE:Flow::ExecutableOp right before stream formation, and in executable serialization (iree/compiler/Dialect/HAL/Target/LLVM/LLVMAOTTarget.cpp:25X) the Flow::ExecutableOp gets called as a parent of the variantOp (not sure if that's how parent ops work), then the sourceIRSummary string attribute is fetched from the parent Flow::ExecutableOp and passed onto the librarybuilder export as a dispatch tag.

I can commit and push all these changes if needed for accurate assessment (it's not all that different from my last public commit), but I've been blocked on trying to get this string passed down through the translation pipeline for a while and could use some direction on whether I'm missing some connecting pieces to this implementation.